tests/scene/test_decal.h

```diff
@@ -0,0 +1,135 @@
+/**************************************************************************/
+/*  test_decal.h                                                          */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef TEST_DECAL_H
+#define TEST_DECAL_H
+
+#include "scene/3d/decal.h"
+
+#include "tests/test_macros.h"
+
+namespace TestDecal {
+TEST_CASE("[SceneTree][Decal] Decal checks get_aabb") {
```

Comment:
Can you restructure your test cases a bit? Looks like only the first two subcases actually test `get_aabb`, the other ones should be separate test cases. 
Alternatively, renaming this test case to "[SceneTree][Decal] Getters/Setters" and the first two subcases would also be fine I think.

Reply:
Could you tell me exactly what I should do with the other two subcases? I am a bit confused because the prompt for this unit test was basically "Call various setters and getters for the class, then check the ```get_aabb``` value".

Reply:
Oh sorry, my bad, I was looking a bit too fast over your code. Something like "[SceneTree][Decal] AABB checks" should be fine then.